linera-chain/src/chain.rs

```diff
@@ -918,12 +933,31 @@ where
             messages.len(),
             block.incoming_bundles.len() + block.operations.len()
         );
-        Ok(BlockExecutionOutcome {
+        let outcome = BlockExecutionOutcome {
             messages,
             state_hash,
             oracle_responses,
             events,
-        })
+        };
+        // Check the size again: Adding the parts could have missed changes in the ULEB128-encoded
+        // vector lengths. If that's the case, we blame the last transaction.
+        let executed_block_size = bcs::serialized_size(&(block, &outcome))
```

Comment:
Recomputing the total size just for a few ULEB128 feels overkill. Vector lengths are bounded in BCS. Can we be conservative and increase the offset instead? (also maybe add a constant for each transaction)

Reply:
Actually, it shouldn't be hard to track the exact size. I'll just account for the ULEB128 encoding whenever we add to a list.